mopidy_mpd/network.py

```diff
@@ -119,19 +119,24 @@ def __init__(
         self.watcher = self.register_server_socket(self.server_socket.fileno())
 
     def create_server_socket(self, host, port):
-        socket_path = get_unix_socket_path(host)
-        if socket_path is not None:  # host is a path so use unix socket
-            sock = create_unix_socket()
-            sock.bind(socket_path)
-        else:
-            # ensure the port is supplied
-            if not isinstance(port, int):
-                raise TypeError(f"Expected an integer, not {port!r}")
-            sock = create_tcp_socket()
-            sock.bind((host, port))
-
-        sock.setblocking(False)
-        sock.listen(1)
+        try:
+            fdnames = os.environ.get("LISTEN_FDNAMES", "").split(":")
+            fd = fdnames.index("mpd") + 3 # got a socket from systemd
```

Comment:
Where does the 3 come from?

Reply:
It's a systemd magic number. The file descriptor for the first socket it hands over is 3, the next one is 4, etc. It's described here https://www.freedesktop.org/software/systemd/man/sd_listen_fds.html#

Reply:
That needs to be detailed as a comment in the code, so it’s always easily available to maintainers.